### PR TITLE
Fixes https://github.com/NuGet/Home/issues/2423.

### PR DESCRIPTION
**Pull request by alpaix**
Fixed VS crash caused by invalid package icon URL.
As reported in the issue having uri with "malformed" scheme triggered
`NotSupportedException`. In order to fix that this exception type was added to the
"whitelist" of errors.

//cc @rrelyea @emgarten @yishaigalatzer

### CODE REVIEW

**Comment by deepakaravindr** [comment]
1. Don't you wanna check this in to dev and cherry-pick into release-3.4.2?
2. Please cherry-pick into release-3.4.2-dev15 branch as well.

File: src/NuGet.Clients/PackageManagement.UI/Converters/IconUrlToImageCacheConverter.cs [hunk 1]
```diff
@@ -75,22 +75,27 @@ public object Convert(object value, Type targetType, object parameter, CultureIn
             }
             // if the URL is a file: URI (which actually happened!), we'll get an exception.
             // if the URL is a file: URI which is in an existing directory, but the file doesn't exist, we'll fail silently.
-            catch (Exception e) when (e is System.IO.IOException || e is System.Net.WebException)
+            catch (Exception e)
+            when (
```
Comment (yishaigalatzer) [inline src/NuGet.Clients/PackageManagement.UI/Converters/IconUrlToImageCacheConverter.cs]: I thought @rrelyea and I commented here, anyways we want to remove the when, and potentially log to activitylog for any unexpeceted exception here

File: test/NuGet.Clients.Tests/NuGet.PackageManagement.UI.Test/Converters/IconUrlToImageCacheConverterTests.cs [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+                DefaultPackageIcon,
+                Thread.CurrentThread.CurrentCulture);
+
+            Assert.Same(DefaultPackageIcon, image);
+        }
+
+        [Fact]
+        public void Convert_WithValidImageUrl_DownloadsImage()
+        {
+            var iconUrl = new Uri("http://go.microsoft.com/fwlink/?LinkID=288859");
```
Comment (yishaigalatzer) [inline test/NuGet.Clients.Tests/NuGet.PackageManagement.UI.Test/Converters/IconUrlToImageCacheConverterTests.cs]: add a test for an invalid image (so a valid http link leading to an html page such as - https://dist.nuget.org/index.html

**Comment by yishaigalatzer** [comment]
:watch:

**Comment by alpaix** [comment]
:runner: As I mentioned offline not all WPF image loading scenarios can be tested with xunit. Will be setting a manual validation procedure.

**Pull request merged**